### PR TITLE
Add dependabot and scheduled runs

### PR DESCRIPTION
**Pull request by Chargothrond**

### CODE REVIEW

**Comment by Chargothrond** [comment]
@riccardoporreca, I tested dependabot and got familiar with it in my forked and also committed the other changes: https://github.com/Chargothrond/ShinyCICD/pull/3 (dependabot and what it suggests are in separate PRs).

If fine with you, I would rebase this branch (I created a bit of a mess as the default behavior of forks is counter-intuitive and too easy to overlook, I guess it's not worth to re-write history for this), do the renv + schedule changes and let dependabot prepare a separate PR for the update of `actions/checkout`.

By next week, we should also see some evidence of the scheduled run on my fork.

**Comment by riccardoporreca** [comment]
@Chargothrond, plaese proceed with the remaining changes to this branch, namely

1. update locked dependencies for renv
2. introduce scheduled runs, once a month should be enough

File: .github/workflows/ci-cd-renv.yml [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+# workflow_dispatch allows you to run this workflow manually from the Actions tab
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+  # ...on a regular schedule for checking existing links turning broken; this is
+  # set to run each 1st of the month to make sure Actions caches are
+  # preserved (they expire every 7 days) and available for future runs
+  schedule:
+    - cron:  '0 0 1 * *'
```
Comment (riccardoporreca) [inline .github/workflows/ci-cd-renv.yml:17]: Rephrased and consistent comments. Note that the `schedule`d runs are only relevant for ShinyCICD and will not end up in `techguides`.

```suggestion
on:
  # Triggered on push and pull request events
  push:
  pull_request:
  # Allow manual runs from the Actions tab
  workflow_dispatch:
  # Monthly runs on the 1st day of the month at midnight
  # (specific for ShinyCICD maintenance)
  schedule:
    - cron:  '0 0 1 * *'
```

Comment (riccardoporreca) [reply .github/workflows/ci-cd-renv.yml:17]: The same applies to `ci-cd.yml`

**Review by riccardoporreca — CHANGES_REQUESTED** [review]
Looks good, just include the improved comments for the triggers

File: .github/workflows/ci-cd-renv.yml (hunk 1 above)
Comment (Chargothrond) [reply .github/workflows/ci-cd-renv.yml]: Updated comments on both files.

**Review by riccardoporreca — APPROVED** [review]
LGTM

**Pull request merged**